### PR TITLE
[T3A4][T09-A2]Clarence Chee

### PR DESCRIPTION
**Pull request by cheec**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -100,19 +102,24 @@ private void recordResult(CommandResult result) {
...
+    private CommandResult executeCommand(Command command) {
+        CommandResult result = null;
         try {
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            result = command.execute();
             storage.save(addressBook);
-            return result;
+        } catch (FileNotFoundException fnfe) {
+            ui.showToUserErrorMsg(fnfe.getMessage());
```
Comment (nirandiw) [inline src/seedu/addressbook/Main.java]: Good place to handle the exception.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -79,12 +79,31 @@ private static boolean isValidPath(Path filePath) {
...
+     * 
+     * @throws StorageOperationException
+     *             if there were errors converting and/or storing data to file.
+     * @throws FileNotFoundException
+     *             if file does not exist in set path.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+        final boolean fileExists = path.toFile().exists();
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java]: - Why final? While running the program the file can go missing so fileExists value can change. For an immutable final variable it won't be possible to change the value.

**Comment by nirandiw** [comment]
@cheec Some comments added. Please close the PR after reading comments

**Comment by cheec** [comment]
@nirandiw doesn't the boolean variable re-instantiate with each function call? if so then there should be only one assignment per call

**Pull request closed without merging**